File: backend/services/bukra_weights.py
"""
Bukra Score category weights.

These weights control how each category contributes to the final Bukra Score.
Category DISPLAY scores (e.g. 21/25) are never affected — only the final 0-100.

Edit only here. Never hardcode weights in bukra_score.py or anywhere else.

Rationale:
  Profitability (0.30)  — Most predictive of long-term business durability.
  Cash Flow    (0.25)   — Real cash is harder to manipulate than reported earnings.
  Stability    (0.20)   — Balance-sheet strength protects against adversity.
  Growth       (0.15)   — Important but already partially captured by profitability trends.
  Debt         (0.10)   — Least independent; captured better by stability metrics.

Sum must equal 1.0.
"""

BUKRA_WEIGHTS: dict[str, float] = {
    "profitability": 0.30,
    "cash_flow":     0.25,
    "stability":     0.20,
    "growth":        0.15,
    "debt":          0.10,
}

# Verify weights sum to 1.0 at import time (fast, no deps)
_total = sum(BUKRA_WEIGHTS.values())
assert abs(_total - 1.0) < 1e-9, (
    f"BUKRA_WEIGHTS must sum to 1.0 — got {_total}. Fix bukra_weights.py."
)

File: frontend/src/config/bukraWeights.ts
/**
 * Bukra Score category weights — frontend mirror of backend/services/bukra_weights.py.
 *
 * The actual scoring computation happens in the Python backend.
 * This file is the single source of documentation for the frontend
 * and can be used to display weight rationale in future UI features.
 *
 * When changing weights: update BOTH files and keep them in sync.
 */
export const BUKRA_WEIGHTS: Record<string, number> = {
  profitability: 0.30,
  cash_flow:     0.25,
  stability:     0.20,
  growth:        0.15,
  debt:          0.10,
}

/**
 * Human-readable rationale per category weight (for future UI display).
 */
export const BUKRA_WEIGHT_RATIONALE: Record<string, { en: string; he: string }> = {
  profitability: {
    en: 'Most predictive of long-term business durability.',
    he: 'הפרמטר המנבא ביותר לאיתנות עסקית לטווח ארוך.',
  },
  cash_flow: {
    en: 'Real cash is harder to manipulate than reported earnings.',
    he: 'תזרים מזומנים אמיתי קשה יותר לניפוח מרווחים מדווחים.',
  },
  stability: {
    en: 'Balance-sheet strength protects against adversity.',
    he: 'איתנות המאזן מגנה בעת משברים.',
  },
  growth: {
    en: 'Important but partially captured by profitability trends.',
    he: 'חשוב, אך נלכד חלקית במגמות הרווחיות.',
  },
  debt: {
    en: 'Least independent; well-captured by stability metrics.',
    he: 'פחות עצמאי — נלכד טוב יותר ע"י מדדי היציבות.',
  },
}
